xrpl4j-crypto-parent/xrpl4j-crypto-core/src/main/java/org/xrpl/xrpl4j/crypto/core/keys/PublicKey.java

```diff
@@ -107,8 +111,8 @@ default String hexValue() {
    * Abstract implementation for immutables.
    */
   @Value.Immutable
-  @JsonSerialize(as = ImmutableDefaultPublicKey.class)
-  @JsonDeserialize(as = ImmutableDefaultPublicKey.class)
+  @JsonSerialize(using = PublicKeySerializer.class)
+  @JsonDeserialize(using = PublicKeyDeserializer.class)
   abstract class DefaultPublicKey implements PublicKey {
```

Comment:
Why do we need a default implementation here, rather than just making `PublicKey` an immutable?